engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDecisionState.java

```diff
@@ -25,55 +26,65 @@
 
 public final class DbDecisionState implements MutableDecisionState {
 
-  private final DbLong dbDecisionKey = new DbLong();
-  private final PersistedDecision dbPersistedDecision = new PersistedDecision();
-  private final DbString dbDecisionId = new DbString();
-
-  private final DbLong dbDecisionRequirementsKey = new DbLong();
-  private final PersistedDecisionRequirements dbPersistedDecisionRequirements =
-      new PersistedDecisionRequirements();
-  private final DbString dbDecisionRequirementsId = new DbString();
-
-  private final DbCompositeKey<DbLong, DbLong> dbDecisionRequirementsKeyAndDecisionKey =
-      new DbCompositeKey<>(dbDecisionRequirementsKey, dbDecisionKey);
-
+  private final DbLong dbDecisionKey;
+
+  private final DbForeignKey<DbLong> fkDecision;
+  private final DbLong dbDecisionRequirementsKey;
+  private final DbForeignKey<DbLong> fkDecisionRequirements;
+  private final DbCompositeKey<DbForeignKey<DbLong>, DbForeignKey<DbLong>>
```

Comment:
:x: please re-add blank lines. These are used to separate the different top level composite keys

Reply:
My bad, not sure why I removed those. Should be fixed now :+1: